huaweicloud/services/css/resource_huaweicloud_css_thesaurus.go

```diff
@@ -0,0 +1,225 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+)
+
+func ResourceCssthesaurus() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: ResourceCssthesaurusCreate,
+		ReadContext:   ResourceCssthesaurusRead,
+		UpdateContext: ResourceCssthesaurusCreate,
+		DeleteContext: ResourceCssthesaurusDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"cluster_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"bucket_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"main_object": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"stop_object": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"synonym_object": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"update_time": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func ResourceCssthesaurusCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+	cssV1Client, err := config.CssV1Client(region)
+	if err != nil {
+		return diag.Errorf("Error creating CSS V1 client: %s", err)
+	}
+	opts := buildThesaurusCreateParameters(d)
+	clusterId := d.Get("cluster_id").(string)
+
+	loadErr := thesaurus.Load(cssV1Client, clusterId, *opts)
+	if loadErr.Err != nil {
+		return diag.Errorf("load thesaurus to css cluster failed. cluster_id=%s,error=%s", clusterId, loadErr.Err)
+	}
+
+	createResultErr := checkThesaurusLoadResult(ctx, cssV1Client, clusterId, d.Timeout(schema.TimeoutCreate))
+	if createResultErr != nil {
+		return diag.FromErr(createResultErr)
+	}
+
+	d.SetId(clusterId)
+
+	return ResourceCssthesaurusRead(ctx, d, meta)
+}
+
+func buildThesaurusCreateParameters(d *schema.ResourceData) *thesaurus.LoadThesaurusReq {
+	opts := thesaurus.LoadThesaurusReq{
+		BucketName: d.Get("bucket_name").(string),
+	}
+
+	if obj, ok := d.GetOk("main_object"); ok {
+		opts.MainObject = obj.(string)
+	}
+	if obj, ok := d.GetOk("stop_object"); ok {
+		opts.StopObject = obj.(string)
+	}
+	if obj, ok := d.GetOk("synonym_object"); ok {
+		opts.SynonymObject = obj.(string)
+	}
+
+	return &opts
+}
+
+func ResourceCssthesaurusRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
`ctx` is unused, pls use `_` instead

Reply:
maybe used later, it's better declare the args name,. so keep it as it is.

Reply:
In GO language, if the variable is not used, it should be declared as _